module/platform/darwin/SensorFilter/src/SensorFilter.cpp

```diff
@@ -264,15 +264,32 @@ namespace module::platform::darwin {
                     covariance.omegaTTt_bias = this->config.motionFilter.initial.covariance.gyroscopeBias;
 
                     // Reset the filter
-                    bool cholesky_success = motionFilter.reset(mean_state.getStateVec(),
-                                                               covariance.asDiagonal(),
-                                                               motionFilter.ALPHA_DEFAULT,
-                                                               motionFilter.KAPPA_DEFAULT,
-                                                               motionFilter.BETA_DEFAULT);
-
-                    // If a Cholesky decomposition failed, wait another MIN_UPDATES and try again
-                    if (!cholesky_success) {
-                        NUClear::log<NUClear::WARN>("Cholesky decomposition failed after filter reset. Retry reset");
+                    Eigen::ComputationInfo cholesky_result = motionFilter.reset(mean_state.getStateVec(),
+                                                                                covariance.asDiagonal(),
+                                                                                motionFilter.ALPHA_DEFAULT,
+                                                                                motionFilter.KAPPA_DEFAULT,
+                                                                                motionFilter.BETA_DEFAULT);
+
+                    // If a Cholesky decomposition failed, log the error, then wait another MIN_UPDATES and try again
+                    if (cholesky_result != Eigen::Success) {
+                        switch (cholesky_result) {
+                            case Eigen::NumericalIssue:
+                                NUClear::log<NUClear::WARN>(
```

Comment:
Inside a reactor you don't need to prefix `log` with `NUClear::` (unless the compiler is getting confused with `std::log` ... in which case it might be better to use `this->log`)